### PR TITLE
adding WAF association to ALB for shared

### PR DESCRIPTION
**Pull request by JamesEElliott**
### Submitter Checklist
-  [X] Is there an impact on Auditing and Logging procedures or capabilities?
-  [ ] Is there an impact on Authentication procedures or capabilities?
-  [ ] Is there an impact on Authorization procedures or capabilities?
-  [ ] Is there an impact on Communication Security procedures or capabilities?
-  [ ] Is there an impact on Cryptography procedures or capabilities?
-  [ ] Is there an impact on Sensitive Data procedures or capabilities?
-  [X] Is there an impact on any other security-related procedures or capabilities?
-  [ ] No security impacts identified.

Security Risks Identified - For any applicable items on the "Submitter Checklist," describe the impact of the change and any implemented mitigations.

This associates the Shared ALB with the WAF.  Prior to this change the Shared ALB had no WAF association and was unsecured.

### CODE REVIEW

File: alb-shared-to-k8s.tf [hunk 1]
```diff
@@ -175,3 +175,14 @@ resource "aws_security_group_rule" "batcave_alb_shared_egress" {
   cidr_blocks       = ["0.0.0.0/0"]
   security_group_id = aws_security_group.batcave_alb_shared[0].id
 }
+
+data "aws_wafv2_web_acl" "cms_waf" {
+  name  = "RegSamQuickACLEnforcingV2"
+  scope = "REGIONAL"
+}
```
Comment (bushong1) [inline alb-shared-to-k8s.tf:182]: This is redundant with the one created in the other file; all .tf files exist in the same context.

File: alb-shared-to-k8s.tf [hunk 2]
```diff
@@ -175,3 +175,14 @@ resource "aws_security_group_rule" "batcave_alb_shared_egress" {
...
+
+resource "aws_wafv2_web_acl_association" "cms_waf_assoc" {
```
Comment (bushong1) [inline alb-shared-to-k8s.tf:184]: Name conflict for the same context reasons
```suggestion
resource "aws_wafv2_web_acl_association" "batcave_alb_shared_cms_waf_assoc" {
```

**Comment by robo-gotham** [comment]
Snyk Scanning for Commit: 8fac67a73406dca784e7bf56be0ebff02aecdd52

Snyk Infrastructure as Code

- Snyk testing Infrastructure as Code configuration issues.
✔ Test completed.

Issues
  No vulnerable paths were found!

-------------------------------------------------------

</details>

Test Summary

  Organization: batcave-ispg
  Project name: CMS-Enterprise/batcave-tf-cluster

✔ Files without issues: 10
✗ Files with issues: 0
  Ignored issues: 0
  Total issues: 0 [ 0 critical, 0 high, 0 medium, 0 low ]

-------------------------------------------------------

Report Complete

  Your test results are available at: https://snyk.io/org/batcave-ispg/projects
  under the name: CMS-Enterprise/batcave-tf-cluster

File: alb-shared-to-k8s.tf (hunk 2 above)
Comment (JamesEElliott) [reply alb-shared-to-k8s.tf:184]: That was lifted verbatim from the alb-public-to-k8s.tf so for naming convention should that be changed too?

Comment (bushong1) [reply alb-shared-to-k8s.tf]: Right, you can't have 2 resources named the same thing.  It's just gotta be different.  Naming convention change would be great, but  that would mean resource migration which is annoying.

**Review by bushong1 — APPROVED** [review]

**Pull request merged**